src/Everscale/Entry.cpp

```diff
@@ -0,0 +1,33 @@
+// Copyright © 2017-2021 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Entry.h"
+
+#include "Address.h"
+#include "Signer.h"
+#include "WorkchainType.h"
+
+using namespace TW::Everscale;
+using namespace TW;
+using namespace std;
+
+// Note: avoid business logic from here, rather just call into classes like Address, Signer, etc.
+
+bool Entry::validateAddress([[maybe_unused]] TWCoinType coin, const string& address, TW::byte, TW::byte, const char*) const {
+    return Address::isValid(address);
+}
+
+string Entry::normalizeAddress([[maybe_unused]] TWCoinType coin, const string& address) const {
+    return Address(address).string();
+}
+
+string Entry::deriveAddress([[maybe_unused]] TWCoinType coin, const PublicKey& publicKey, TW::byte, const char*) const {
+    return Address(publicKey, WorkchainType::Basechain).string();
+}
+
+void Entry::sign([[maybe_unused]] TWCoinType coin, const TW::Data& dataIn, TW::Data& dataOut) const {
+    signTemplate<Signer, Proto::SigningInput>(dataIn, dataOut);
+}
```

Comment:
Instead of `using namespace TW::Everscale` please integrate it into the same namespace:

```cpp
namespace TW::Everscale {
  // my code
}
```